src/modules/map/components/olMap/handler/featureInfo/OlFeatureInfoHandler.js

```diff
@@ -74,8 +77,19 @@ export class OlFeatureInfoHandler extends OlMapHandler {
 					data: { geometry: featureInfo.geometry.data, geometryType: HighlightGeometryTypes.GEOJSON }
 				}));
 
-			//publish current HighlightFeature items
-			addHighlightFeatures(highlightFeatures);
+			const unsubscribe = observe(this._storeService.getStore(), state => state.featureInfo.querying, querying => {
+				//untestable else path cause function is self-removing
+				/* istanbul ignore else */
+				if (!querying) {
+					//publish current HighlightFeature items when all pending queries are resolved
+					addHighlightFeatures(highlightFeatures);
+					unsubscribe();
+				}
+			});
```

Comment:
a possible refactoring for the future, if this occur more often: `observeOnce` mainly to prevent, to have too much ignore flags

Reply:
Good point!